src/components/MAPI/clusters/ClusterDetail/ClusterDetailOverview.tsx

```diff
@@ -34,15 +34,18 @@ const ClusterDetailOverview: React.FC<{}> = () => {
   const { orgId, clusterId } = match.params;
 
   const clientFactory = useHttpClientFactory();
+  const orgClientFactory = useHttpClientFactory();
+
   const auth = useAuthProvider();
 
   const clusterClient = useRef(clientFactory());
+  const orgClient = useRef(orgClientFactory());
```

Comment:
You can re-use the same factory. `clientFactory()` will create a new instance of `HttpClientImpl`, so the mutation thingy is not a problem anymore. It's just a matter of not using the same instance of `HttpClientImpl`

```suggestion
  const orgClient = useRef(clientFactory());
```